cmd/pinniped-proxy/src/https.rs

```diff
@@ -73,11 +80,56 @@ pub fn cert_for_cert_data(cert_data: Vec<u8>) -> Result<Certificate> {
     }
 }
 
+/// rewrite_request directs the specified request to the targeted backend api
+/// server.
+pub fn rewrite_request(mut req: Request<hyper::Body>, k8s_api_server_url: String) -> Result<Request<hyper::Body>> {
+    // Update the request URI.
+    let uri_string = format!(
+        "{}{}",
+        k8s_api_server_url,
+        req.uri().path_and_query().map(|x| x.as_str()).unwrap_or("")
```

Comment:
I guess this part is for concatenating `k8s_api_server_url` to the rest of the request uri, but,just for the sake of clarity, would it be a good idea to add an example as a comment so that we can get a better idea of the big picture?

Reply:
Yep, will do.